wallet/src/account/mod.rs

```diff
@@ -1157,12 +1188,35 @@ impl Account {
         self.key_chain.get_addresses_usage_state()
     }
 
-    /// Return true if this transaction output is can be spent by this account or if it is being
+    fn collect_output_destinations(&self, txo: &TxOutput) -> Vec<Destination> {
```

Comment:
This function puts us back at the "every output has one destination" and "every output has a some 'value' associated with it", which is what we've trying to avoid everywhere, and which is why TxOutput doesn't have such getters. While this function is called "collect_output_destinations", we're actively preferring decommission destinations instead of staking destinations. Now I understand why you're doing this, but this doesn't rhyme with the name of the function or how it works based on output types. Please take another look at this.

Reply:
I'm not sure I understand your concern. The statement of this function is "every output can have [0;n] destinations". And it is valid only for this specific case in the wallet, where we want to watch all destinations of an output. I'm not preferring any destination here and just want to collect everything related.

Reply:
if we are going to include decommission key to be tracked, we will need a filter on the list pool ids to exclude pools in the case the Staker is not ours when called from the staking code.

Reply:
> we will need a filter on the list pool ids

added the filter, thanks for noticing